ext/user_request.c

```diff
@@ -14,7 +13,7 @@ static struct {
 
 DDTRACE_PUBLIC bool ddtrace_user_req_add_listeners(ddtrace_user_req_listeners *listeners)
 {
-    if (strcmp(sapi_module.name, "cli") != 0) {
+    if (strcmp(sapi_module.name, "cli") != 0 && strcmp(sapi_module.name, "frankenphp") != 0) {
```

Comment:
You know that the frankenphp SAPI has two different ways of working, one worker mode and one which essentially works like fpm? In the latter case you likely want this function to not be skipped.

Reply:
The condition does the opposite: it skips adding the listeners if not on cli (or frankenphp).

If we're running in classic mode, this has no ill effects: `frankenphp_handle_request` will never be called; the frankenphp integration won't be installed and `\DDTrace\UserRequest\notify_start` will never be called; the installed listeners will never be called.

I've added support and tests for classic mode; however, because I'm unable to detect it, it is being the option `DD_APPSEC_CLI_START_ON_RINIT`, which disables the "user request" lifecycle and delineates the request with RINIT/RSHUTDOWN

Reply:
Ah right, this code runs in MINIT, and then you don't know yet. You can only detect that in RINIT via contents of $_SERVER.